### PR TITLE
Issue 6 - add new reminder from /new

### LINKED ISSUE #6: Create A New Reminder
At the bottom of the `reminders.hbs` template, there should be a button that routes the user to `'/new'` where users can create a new reminder.

* The new reminder should have inputs for the title, date, and notes.
* The new reminder should have a button for submitting the new note.
* Pressing the button, should submit reminder to the database and have it shown in the Reminders list.

**Hot Tip**: TDD is really helpful.

### PR DESCRIPTION
**Pull request by rcwestlake**
This feature adds a new reminder to the existing list from the /new url. We created a new component that has an action, which sets the properties to match the value of the input fields and then creates a record in the store. 

@martensonbj @tman22 @stevekinney @nfosterky

We wrote tests to confirm that this is working. 

This closes #6.

### CODE REVIEW

File: app/components/new-reminder.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  store: Ember.inject.service(),
+
+  tagName: 'section',
```
Comment (nfosterky) [inline app/components/new-reminder.js]: 👍  for using tagName

File: app/router.js [hunk 2]
```diff
@@ -9,6 +9,7 @@ const Router = Ember.Router.extend({
 Router.map(function() {
   this.route('reminders', function() {
     this.route('reminder', { path: ':id'});
+    this.route('new', { path: '/new'})
```
Comment (nfosterky) [inline app/router.js]: `{ path: '/new'}` is redundant since the route is already `'new'`

File: app/templates/components/new-reminder.hbs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  </label>
+  <label>
+    Date
+    {{input type="date" class="spec-date-field" value=date}}
+  </label>
+  <label>
+    Notes
+    {{textarea class="spec-notes-field" value=notes}}
+  </label>
+  <button class="add-reminder--submit">Save Reminder</button>
```
Comment (nfosterky) [inline app/templates/components/new-reminder.hbs]: fix your classname

File: app/templates/reminders.hbs [hunk 4]
```diff
@@ -12,4 +12,9 @@
   {{/each}}
 </ul>
 
+<button>
+  {{#link-to 'reminders.new' class='spec-new-reminder'}}
+    add reminder
```
Comment (nfosterky) [inline app/templates/reminders.hbs]: Capitalize button titles

File: app/templates/reminders/new.hbs [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{{new-reminder}}
+
+{{outlet}}
```
Comment (nfosterky) [inline app/templates/reminders/new.hbs]: Delete `{{outlet}}` unless you use it

File: app/templates/components/new-reminder.hbs (hunk 3 above)
Comment (rcwestlake) [inline app/templates/components/new-reminder.hbs]: @nfosterky what class name are you referring to?

**Comment by rcwestlake** [comment]
@nfosterky Changes have been made. Thanks for the input.

File: tests/unit/routes/new-reminder-test.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { moduleFor, test, skip } from 'ember-qunit';
```
Comment (martensonbj) [inline tests/unit/routes/new-reminder-test.js]: Let's get rid of this test and add an acceptance test instead that checks for functionality of a user adding a new reminder.

File: tests/unit/routes/reminder-test.js [hunk 7]
```diff
@@ -1,11 +1,11 @@
-import { moduleFor, test } from 'ember-qunit';
```
Comment (martensonbj) [inline tests/unit/routes/reminder-test.js]: We don't need this test either if you're not using it.

File: tests/acceptance/reminder-list-test.js [hunk 8]
```diff
@@ -1,20 +1,28 @@
...
+
 test('viewing the homepage', function(assert) {
   server.createList('reminder', 5);
 
   visit('/');
 
   andThen(function() {
-    assert.equal(currentURL(), '/reminders');
-    assert.equal(Ember.$('.spec-reminder-item').length, 5);
+    assert.equal(currentURL(), '/reminders', 'visits the /reminders page');
```
Comment (rcwestlake) [inline tests/acceptance/reminder-list-test.js]: I added a summary sentence for context in our test server

**Comment by rcwestlake** [comment]
@martensonbj @nfosterky 

We removed unnecessary test files. The request for test coverage on the new reminders is already there. See Tests > Acceptance > new-reminder-test.js

File: tests/acceptance/reminder-list-test.js (hunk 8 above)
Comment (martensonbj) [inline tests/acceptance/reminder-list-test.js]: 👍

**Comment by martensonbj** [comment]
Looks good - sorry for the oversight on the existing test.

**Pull request merged**